packages/@react-aria/menu/src/utils.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {Key} from '@react-types/shared';
+import {TreeState} from '@react-stately/tree';
+
+interface MenuData {
+  onClose?: () => void,
+  onAction?: (key: Key) => void,
+  shouldUseVirtualFocus?: boolean,
+  id: string
+}
+
+export const menuData = new WeakMap<TreeState<unknown>, MenuData>();
+
+function normalizeKey(key: Key): string {
+  if (typeof key === 'string') {
+    return key.replace(/\s*/g, '');
+  }
+
+  return '' + key;
+}
+
+export function getItemId<T>(state: TreeState<T>, itemKey: Key): string {
+  let data = menuData.get(state);
+
+  if (!data) {
+    throw new Error('Unknown menu');
+  }
+
+  return `${data.id}-option-${normalizeKey(itemKey)}`;
```

Comment:
i guess that is true but if we keep the spaces in the key, then the DOM id generated is invalid no? As for `5` and `'5'` do we support that technically? I guess we could generate a completely separate DOM `id` and sustain a mapping between the collection keys and their respective DOM `ids` but the `data-key` is always a string and matches the id the user provides

Reply:
I think that's what we should do then, because yeah, the spaces would be invalid in the dom

Reply:
I've gone ahead and removed this normalization completely since the menu items communicate their ids upwards via the custom events. For components like TableView/ComboBox/etc that have always historically done this normalization, we can look into updating them separately perhaps